agents/src/utils.ts

```diff
@@ -362,24 +389,59 @@ export class AbortableTask<T> {
       });
   }
 
+  /**
+   * Cancels the task.
+   */
   cancel() {
     this.controller.abort();
   }
 
+  /**
+   * Cancels the task and waits for it to complete.
+   *
+   * @param timeout - The timeout in milliseconds
+   * @returns The result of the task
+   * @throws {@link TASK_TIMEOUT_ERROR} if the task times out
```

Comment:
If you want to be explicit about errors being thrown, then keep in mind that you're currently throwing _all_ errors that happen in `await result` additionally to the timeout error